libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -202,10 +204,11 @@ void StructuralSimplifier::simplify(std::vector<yul::Statement>& _statements)
 		_statements,
 		[&](Statement& _stmt) -> OptionalStatements
 		{
-			visit(_stmt);
```

Comment:
The problem here was that if we visit the statement before we modify it, it will invoke the data flow analyzer on that statement and potentially assign future variable values to m_values.